[TASK] Add support for issueing PKI Certificates

UXabre opened the pull request:
I've added the ability to issue certificates when using a PKI secrets engine.
I've purposely kept the options undefined, as the PKI is a very versatile engine. The only thing that might bother me, is the path, being "<pki_engine_name>/issue/<pki_role>" is to be used to issue a request

ekohl commented on README.md:
```diff
@@ -33,13 +39,23 @@ As result you should get secret data, e.g.
 ## Copyright
 
-Copyright (c) 2018 dmTECH GmbH, [dmtech.de](https://www.dmtech.de/)
+Copyright (c) 2020 dmTECH GmbH, [dmtech.de](https://www.dmtech.de/)
```
Shouldn't this be 2018-2020?

UXabre replied:
Oh yes, sorry about that!

timogoebel commented on README.md:
```diff
@@ -1,3 +1,4 @@
+
```
Nit: can we remove this empty line?

UXabre replied:
Yes, I'll remove it (sorry, usually I notice these things but I overlooked it this time)

timogoebel replied:
This line still wants to be removed.

kamils-iRonin commented on README.md:
````diff
@@ -33,13 +39,23 @@ As result you should get secret data, e.g.
+
+
+```
+<%= issue_certificate('MyVault', 'pkiEngine/issue/testRole', common_name: 'test.mydomain.com', ttl: '10s') %>
````
`vault_issue_certificate` is the correct macro name, right?

UXabre replied:
Corrected it! Thanks for spotting this!

UXabre commented:
I'll add tests latter tonight! Thanks for reviewing so far!

UXabre commented:
I've added a test, as well as adding the macro to the safe functions of the template function

thanks @kamils-iRonin, @timogoebel & @ekohl for the reviews so far :-)

UXabre commented:
Hi @timogoebel ,

I overlooked the white line :-( sorry! But it's resolved now! Thanks for your review!

timogoebel commented:
@UXabre: Looks great, thanks! I'll see to it that we fix the CI pipeline here. Then we can rebase this PR and merge it when tests pass.

timogoebel commented:
@UXabre: Do you mind rebasing this?

UXabre commented:
Hi @timogoebel ,

Rebased and ready to go! :-) Thanks for fixing the test suite! I noticed some rubocop issues and fixed them as well.

Thanks for and happy reviewing!

timogoebel commented:
Thanks, @UXabre.

timogoebel commented:
@UXabre: I've just released version 0.1.0 of this gem that contains these changes!
